graphql/src/parser/parse_query.rs

```diff
@@ -70,15 +70,23 @@ fn parse_value(stream: &mut TokenStream) -> QlResult<Value> {
     stream.expect(maybe_parse_value)
 }
 
-// Name (args)? { field list }?
+// Name (: Name)? (args)? { field list }?
 fn maybe_parse_field(stream: &mut TokenStream) -> QlResult<Option<Field>> {
-    let name = none_ok!(maybe_parse_name(stream)?);
+    let name_or_alias = none_ok!(maybe_parse_name(stream)?);
+    let (alias, name) = match none_ok!(stream.peek_tok()).kind {
+        TokenKind::Atom(Atom::Colon) => {
+            stream.bump();
+            (Some(name_or_alias), none_ok!(maybe_parse_name(stream)?))
```

Comment:
I think here you want `parse_name(stream)?` rather than `none_ok!(maybe_parse_name(stream)?)`

Reply:
@nrc done.